paperqa/agents/models.py

```diff
@@ -0,0 +1,520 @@
+                )
+            v = STATIC_PROMPTS[values["named_prompt"]]
+        if values["summary_llm"] == "none":
+            v.skip_summary = True
+            # for simplicity (it is not used anywhere)
+            # so that Docs doesn't break when we don't have a summary_llm
+            values["summary_llm"] = "gpt-4o-mini"
+        return v
+
+    def set_docs_name(self, docs_name: str) -> None:
+        """Set the internal docs name for tracking."""
+        self._docs_name = docs_name
+
+    @staticmethod
+    def get_index_name(
+        paper_directory: str | os.PathLike,
+        embedding: str,
+        parsing_configuration: ParsingConfiguration,
+    ) -> str:
+
+        # index name should use an absolute path
+        # this way two different folders where the
+        # user locally uses '.' will make different indexes
+        if isinstance(paper_directory, Path):
+            paper_directory = str(paper_directory.absolute())
+
+        index_fields = "|".join(
+            [
+                str(paper_directory),  # cast for typing
+                embedding,
+                str(parsing_configuration.chunksize),
+                str(parsing_configuration.overlap),
+                parsing_configuration.chunking_algorithm,
+            ]
+        )
+
+        return f"pqa_index_{hexdigest(index_fields)}"
+
+
+class AnswerResponse(BaseModel):
+    answer: Answer
+    usage: dict[str, list[int]]
+    bibtex: dict[str, str] | None = None
+    status: AgentStatus
+    timing_info: dict[str, dict[str, float]] | None = None
+    duration: float = 0.0
+    # A placeholder for interesting statistics we can show users
+    # about the answer, such as the number of sources used, etc.
+    stats: dict[str, str] | None = None
+
+    @field_validator("answer")
+    def strip_answer(
+        cls, v: Answer, info: ValidationInfo  # noqa: ARG002, N805
+    ) -> Answer:
+        # This modifies in place, this is fine
+        # because when a response is being constructed,
+        # we should be done with the Answer object
+        v.filter_content_for_user()
+        return v
+
+    async def get_summary(self, llm_model="gpt-4-turbo") -> str:
+        sys_prompt = (
+            "Revise the answer to a question to be a concise SMS message. "
+            "Use abbreviations or emojis if necessary."
+        )
+        model = OpenAILLMModel(config={"model": llm_model, "temperature": 0.1})
+        chain = model.make_chain(
+            AsyncOpenAI(), prompt="{question}\n\n{answer}", system_prompt=sys_prompt
+        )
+        result = await chain({"question": self.answer.question, "answer": self.answer.answer})  # type: ignore[call-arg]
+        return result.text.strip()
+
+
+class TimerData(BaseModel):
+    # NOTE: use lambda because asyncio.get_running_loop can throw: RuntimeError: no running event loop
+    start_time: float = Field(
+        default_factory=lambda: asyncio.get_running_loop().time()  # noqa: FURB111
+    )
```

Comment:
What do you think of having this `start_time` not come from `asyncio`, but normal `time.time`?

Reply:
good call, i'm not sure why it defaults to this anyway.